Migrators/Helpers/batchLoader.py

```diff
@@ -1,7 +1,25 @@
+from functools import partial
+from multiprocessing.dummy import Pool as ThreadPool
+from tqdm import tqdm
+
 from typedb.client import TransactionType
 
-def write_batch(session, batch):
+def write_batches(session, queries, batch_size, num_threads):
+    with ThreadPool(num_threads) as pool:
+        with tqdm(total=len(queries)) as pbar:
+            chunk_size = max(1, len(queries) // num_threads // batch_size // 10)
+            for n in pool.imap_unordered(partial(write_batch, session), iter_batches(queries, batch_size), chunk_size):
```

Comment:
`imap` avoids copying the entire list, and the `_unordered` variant means it won't stall the progressbar by waiting for a slow worker to finish. Grouping the batches in large chunks helps avoid excessive context switching.